tests/lib/utils

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# Utilities for BATS tests.
+
+function improver_check_skip_acceptance {
+  if [[ -z "${IMPROVER_ACC_TEST_DIR:-}" ]]; then
+    skip "Acceptance test directory not defined"
+  fi
+  if ! type -f nccmp 1>/dev/null 2>&1; then
+    skip "nccmp not installed"
+  fi
+}
+
+function improver_compare_output {
+  # Run nccmp to compare the output and kgo.
+  run nccmp -dmNs "$1" "$2"
```

Comment:
I think that it would be informative to add what the expected output from `nncmp` is. As far as I gather, if the files are identical, it reports: `Files "kgo.nc" and "output.nc" are identical.`, whilst if the files differ, it would print out something like: `DIFFER : VARIABLE "thickness_of_rainfall_amount" IS MISSING ATTRIBUTE WITH NAME "_FillValue" IN FILE "output.nc"`. I think this might help to explain why you're doing the `[[ "$output" =~ "are identical." ]]` test.